logic.py

```diff
@@ -116,11 +122,20 @@ def get_paste_listing(config):
         return "ERROR", "Paste listing has been disabled by the " +\
             "administrator.", 503
 
+    b = config['b']
+
     try:
-        paste_list = config['b'].get_all_paste_ids()
-    except config['b'].e.ErrorException as errmsg:
+        paste_list = b.get_all_paste_ids()
+    except b.e.ErrorException as errmsg:
         return "ERROR", errmsg, 500
 
+    if (paste_list[0] != "none"):
+        # filter pastes to keep the public ones only
+        paste_list = [ paste for paste in paste_list 
```

Comment:
This might be wise, yes. A more generic handling of the paste metadata (visibility and future features such as syntax highlighting) both when creating them and when listing/reading them.